docs/ADR-5-cardano-testnet-node-configuration-file.md

```diff
@@ -0,0 +1,42 @@
+# Status
+
+📜 Proposed 2024-09-10
+
+# Context
+
+There is no way to check a node configuration file for basic sanity issues.
+
+# Decision
+
+## Add a new command in the CLI
+
+Introduce a new `debug check-node-configuration FILEPATH` command in `cardano-cli`. It will have the following options:
+
+```
+--node-configuration-file FILEPATH
```

Comment:
I would avoid listing the command names and option names. Those are parts of the user interface design, and not really n part of an architecture. Additionally, the commands and option names can change during the time and the ADR will be out of sync.

If you insist on keeping the command and option names, can you rephrase the text to make them examples rather than a references to a command option / toggle?
For example instead of

>### Have a `--fix` flag
>
>We considered having a `--fix` flag that would make `check-node-configuration`:

write

>### Provide fix up capability
>
>We considered having a dedicated flag (e.g. `--fix`) that would make the command:

Reply:
@carbolymer> good point :+1:

I addressed your concerns in the additional commit `Do not give exact API in the ADR. Make it more high-level.`